internal/targetallocator/taresources/desired_state.go

```diff
@@ -274,8 +289,25 @@ func assembleClusterRoleBinding(c *targetAllocatorConfig) *rbacv1.ClusterRoleBin
 	}
 }
 
-func assembleService(c *targetAllocatorConfig) *corev1.Service {
-	return &corev1.Service{
+func assembleService(c *targetAllocatorConfig, extraConfig util.ExtraConfig) *corev1.Service {
+	ports := []corev1.ServicePort{
+		{
+			Name:       "http",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       80,
+			TargetPort: intstr.FromString("http"),
+		},
+	}
+
+	if extraConfig.TargetAllocatorMtlsEnabled {
+		ports = append(ports, corev1.ServicePort{
+			Name:       "https",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       443,
+			TargetPort: intstr.FromString("https")})
+	}
```

Comment:
_question:_ If mTLS is enabled, will only port 443 be used? Should we still list port 80 for non-TLS traffic additionally in that setup, or should we rather only have port 443?

In particular, would the scrape configs including the auth information be made available via port 80 right now and would that allow bypassing mTLS?

Reply:
Oh, that's smart 👍